content/tutorials/trainapp-fitbit.textile

```diff
@@ -0,0 +1,531 @@
+     });
+
+     // We use the Ably TFL Hub, to recieve update for the given line & stationid (station & towards)   
+     let channelName = `[product:ably-tfl/tube]tube:${line}:${stationId[0].stationid}:arrivals`;
+     let trainChannel = realtime.channels.get(channelName);
+     // Replace the space with plus so that it can be passed to Google API
+     let dest = station.split(" ").join("+");
+     if (latitude !== undefined && longitude !== undefined) {
+       // Sending current location & station as destination to Google API to get walking time
+       let googleUrl = `https://maps.googleapis.com/maps/api/distancematrix/json?origins=${latitude},${longitude}&destinations=${dest}&mode=walking&key=${apiKey}`;
+       fetch(googleUrl, {
+         method: "GET"
+       })
+         .then(function(res) {
+           return res.json();
+         })
+         .then(function(data) {
+           let myData = data;
+            // Convert the time to mins
+           walkingTime = Math.floor(
+             myData.rows[0].elements[0].duration.value / 60
+           );
+         })
+         .catch(err => console.log("[FETCH]: " + err));
+     }
+
+     // We subscribe to the messages coming on the Ably TFL Hub channel   
+     trainChannel.subscribe(msg => {
+       /* station update in msg */
+       // Get upcoming trains for our user input of line, station and towards
+       const trains = msg.data.filter(train => {
+         return (
+           line === train.LineId &&
+           station === train.StationName &&
+           towards === train.Towards &&
+           new Date(train.ExpectedArrival) > new Date()
+         );
+       });
+       // Get the first upcoming train time
+       let trainTime = new Date(trains[0].ExpectedArrival);
+       let currentTime = new Date();
+       let diff = trainTime - currentTime;
+       timeToArrival = Math.floor(diff / 1000 / 60);
+       // Now using the messaging module we send the data to the app 
+       // Walk time, train time 
+       if (
+         messaging.peerSocket.readyState === messaging.peerSocket.OPEN &&
+         walkingTime !== undefined
+       ) {
+         messaging.peerSocket.send({
+           tTA: timeToArrival,
+           wT: walkingTime,
+           lT: timeToArrival - walkingTime
+         });
+       }
+     });
+}
+```
+
+"Channels":https://www.ably.io/documentation/realtime/channels#title are the medium through which messages are distributed; clients attach to channels to subscribe to messages, and every message published to that channel is broadcast by Ably to all its subscribers.
+
+"See this step in Github":https://github.com/ably/tutorials/commit/55d2b927547b6b7520fe4bde2dec0fa0948ae152
+
+h2(#send-message). Step 7 - Update the settings options onchange of a setting
+
+Now we need to update the settings options(dropdown) on change of any setting since all the settings depend on the previous above settings value.
```

Comment:
Now we need to update the settings options so that their respective dropdowns reflect the value selected in the setting above.